src/main/java/appeng/tile/misc/TileCondenser.java

```diff
@@ -223,65 +241,174 @@ public boolean canExtractItem( final int slotIndex, final ItemStack extractedIte
 	{
-		if( doFill )
-		{
-			this.addPower( ( resource == null ? 0.0 : (double) resource.amount ) / 500.0 );
-		}
-
-		return resource == null ? 0 : resource.amount;
+		this.addPower( 0 );
 	}
 
 	@Override
-	public FluidStack drain( final EnumFacing from, final FluidStack resource, final boolean doDrain )
+	public IConfigManager getConfigManager()
 	{
-		return null;
+		return this.cm;
 	}
 
-	@Override
-	public FluidStack drain( final EnumFacing from, final int maxDrain, final boolean doDrain )
+	public double getStoredPower()
 	{
-		return null;
+		return this.storedPower;
 	}
 
-	@Override
-	public boolean canFill( final EnumFacing from, final Fluid fluid )
+	private void setStoredPower( final double storedPower )
 	{
-		return true;
+		this.storedPower = storedPower;
 	}
 
 	@Override
-	public boolean canDrain( final EnumFacing from, final Fluid fluid )
+	public boolean hasCapability( Capability<?> capability, EnumFacing facing )
 	{
-		return false;
+		if( capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY )
+		{
+			return true;
+		}
+		else if( capability == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY )
+		{
+			return true;
+		}
+		else if( capability == Capabilities.STORAGE_MONITORABLE_ACCESSOR )
+		{
+			return true;
+		}
+		return super.hasCapability( capability, facing );
 	}
 
+	@SuppressWarnings( "unchecked" )
 	@Override
-	public FluidTankInfo[] getTankInfo( final EnumFacing from )
+	public <T> T getCapability( Capability<T> capability, @Nullable EnumFacing facing )
 	{
-		return EMPTY;
+		if( capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY )
+		{
+			return (T) itemHandler;
+		}
+		else if( capability == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY )
+		{
+			return (T) fluidHandler;
+		}
+		else if( capability == Capabilities.STORAGE_MONITORABLE_ACCESSOR )
+		{
+			return (T) meHandler;
+		}
+		return super.getCapability( capability, facing );
 	}
 
-	@Override
-	public void updateSetting( final IConfigManager manager, final Enum settingName, final Enum newValue )
+	private class ItemHandler implements IItemHandler
 	{
-		this.addPower( 0 );
-	}
 
-	@Override
-	public IConfigManager getConfigManager()
-	{
-		return this.cm;
+		@Override
+		public int getSlots()
+		{
+			// We only expose the void slot
+			return 1;
+		}
+
+		@Override
+		public ItemStack getStackInSlot( int slot )
+		{
+			// The void slot never has any content
+			return null;
+		}
+
+		@Override
+		public ItemStack insertItem( int slot, ItemStack stack, boolean simulate )
+		{
+			if( slot != 0 )
+			{
+				return stack;
+			}
+			if( !simulate && stack != null )
+			{
+				addPower( stack.stackSize );
+			}
+			return null;
+		}
+
+		@Override
+		public ItemStack extractItem( int slot, int amount, boolean simulate )
+		{
+			return null;
+		}
 	}
 
-	public double getStoredPower()
+
+	private static final IFluidTankProperties[] EMPTY = { new FluidTankProperties( null, 10, true, false ) };
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1213)